[pkg] add zipfile unit tests

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74929
* #72237

Add unit tests for zipfile classes

PaliC committed:
[pkg] add zipfile unit tests

[ghstack-poisoned]

d4l3k commented:
lint is failing otherwise looks good

d4l3k commented on test/package/test_zipfile.py:
```diff
@@ -0,0 +1,128 @@
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ZipfileWriter = TorchScriptPackageZipFileWriter
+        self.ZipfileReader = TorchScriptPackageZipFileReader
```
should this be using TorchScript one or the not torchscript one?

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

d4l3k reviewed (approved):
reapproving

PaliC committed:
Update on "[pkg] add zipfile unit tests"


Add unit tests for zipfile classes

Differential Revision: [D35254715](https://our.internmc.facebook.com/intern/diff/D35254715)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35254715).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423078).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423078).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423078).
